setup/src/Magento/Setup/Model/Installer/Event/BeforeModuleIsProcessed.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace Magento\Setup\Model\Installer\Event;
+
+use Magento\Framework\Setup\ModuleContextInterface;
+use Magento\Framework\Setup\SetupInterface;
+use Zend\EventManager\Exception\InvalidArgumentException;
+
+/**
+ * Event fired during setup before each module is processed
+ */
+class BeforeModuleIsProcessed extends AbstractInstallerEvent
```

Comment:
We need to re-work on this event name. The current name does not reflect the application event enough.